api/src/main/java/org/openmrs/api/impl/UserServiceImpl.java

```diff
@@ -102,7 +104,12 @@ public User createUser(User user, String password) throws APIException {
 	@Override
 	@Transactional(readOnly = true)
 	public User getUser(Integer userId) throws APIException {
-		return dao.getUser(userId);
+		User user = dao.getUser(userId);
```

Comment:
Hi @wluyima
Could please explain to me what a search method or point one out for me. Please!

Reply:
This method should be able to return any user, it's somehow different from those that return a list of users

Reply:
I don't whether I'm just a newbie but to my little understanding @wluyima , this already looks like a search methods otherwise, do you mind pointing me to one example just for clarity?

Reply:
I'm saying getting a user by an identifier like an Id and uuid should return any user even if it's daemon, this is because it's needed in some areas of the code, otherwise other methods like getUsers(String searchPhrase...) and any search related method should exclude daemon user.

Reply:
@samuelmale search methods return multiple results i.e a collection of hits whereas getUser(userId) and getUser(uuid) return a single user that matches the identifier.

Reply:
Ooooohhh, I have now understood the concept here. Thanks!